lang/cs/Org.Apache.REEF.Network/Elastic/Driver/Default/DefaultElasticTaskSetManager.cs

```diff
@@ -0,0 +1,1344 @@
+            /// </summary>
+            public TaskState TaskStatus { get; private set; }
+
+            /// <summary>
+            /// How many times the task have been scheduled.
+            /// </summary>
+            public int NumRetry = 1;
+
+            /// <summary>
+            /// Save the reference to the remote running task.
+            /// </summary>
+            /// <param name="taskRunner">The reference to the remote running task</param>
+            public void SetTaskRunner(IRunningTask taskRunner)
+            {
+                TaskRunner = taskRunner;
+                _isTaskDisposed = false;
+            }
+
+            /// <summary>
+            /// Change the status of the task.
+            /// </summary>
+            /// <param name="status">The new task state</param>
+            public void SetTaskStatus(TaskState status)
+            {
+                TaskStatus = status;
+            }
+
+            /// <summary>
+            /// Update the task runtime.
+            /// </summary>
+            /// <param name="newActiveContext">The active context of the task</param>
+            /// <param name="evaluatorId">The id of the evaluator</param>
+            public void UpdateRuntime(IActiveContext newActiveContext, string evaluatorId)
+            {
+                if (!_isActiveContextDisposed)
+                {
+                    throw new IllegalStateException("Updating Task with not disposed active context");
+                }
+
+                ActiveContext = newActiveContext;
+                EvaluatorId = evaluatorId;
+                _isActiveContextDisposed = false;
+            }
+
+            /// <summary>
+            /// Set the task runtime as diposed.
+            /// </summary>
+            public void DropRuntime()
+            {
+                _isActiveContextDisposed = true;
+                _isTaskDisposed = true;
+            }
+
+            /// <summary>
+            /// Dipose the task.
+            /// </summary>
+            public void DisposeTask()
+            {
+                if (!_isTaskDisposed)
+                {
+                    TaskRunner?.Dispose();
+
+                    _isTaskDisposed = true;
+                }
+            }
+
+            /// <summary>
+            /// Dipose the active context of the task.
+            /// </summary>
+            public void DisposeActiveContext()
+            {
+                if (!_isActiveContextDisposed)
+                {
+                    ActiveContext?.Dispose();
+
+                    _isActiveContextDisposed = true;
+                }
+            }
+
+            /// <summary>
+            /// Dipose the task info.
+            /// </summary>
+            public void Dispose()
+            {
+                if (!_isDisposed)
+                {
+                    DisposeTask();
+
+                    DisposeActiveContext();
+
+                    _isDisposed = true;
```

Comment:
We should discuss on this. Disposing and dropping runtimes are different. You drop runtimes when the tasks fail, while you dispose them when the task manager finishes / fails. I think that more than removing `_isDisposed` I should add it in other places so that we cannot add a runtime to a disposed task for instance.